File: product/types/app_context.go
package types

import (
	"github.com/autobar-dev/services/product/repositories"
	filerepository "github.com/autobar-dev/shared-libraries/go/file-repository"
)

type Repositories struct {
	Product     *repositories.ProductRepository
	SlugHistory *repositories.SlugHistoryRepository
	Cache       *repositories.CacheRepository
	Meili       *repositories.MeiliRepository
	File        *filerepository.FileRepository
}

type AppContext struct {
	MetaFactors  *MetaFactors
	Config       *Config
	Repositories *Repositories
}

File: emailtemplate/src/types/appContext.ts
import { Config } from "./config";
import { MetaFactors } from "./meta";
import { TemplateRepository } from "../repositories/TemplateRepository";

export type AppContext = {
  metaFactors: MetaFactors,
  config: Config,
  repositories: {
    template: TemplateRepository,
  },
};
